File: lib/models/medication.dart
class Medication {
  final int id;
  final String name;
  final String description;
  final double price;
  final int stock;
  final String type;
  final String category;
  final String? imageUrl;

  Medication({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.stock,
    required this.type,
    required this.category,
    this.imageUrl,
  });

  factory Medication.fromJson(Map<String, dynamic> json) {
    return Medication(
      id: json['id'] as int,
      name: json['name'] as String? ?? '',
      description: json['description'] as String? ?? '',
      price: (json['price'] is int)
          ? (json['price'] as int).toDouble()
          : (json['price'] as num?)?.toDouble() ?? 0.0,
      stock: (json['stock'] ?? json['stock_quantity'] ?? 0) as int,
      type: json['medication_type'] ?? json['type'] ?? 'human',
      category: json['category_name'] ?? json['category'] ?? 'Other',
      imageUrl: json['image_url'] ?? json['imageUrl'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'price': price,
      'stock': stock,
      'type': type,
      'category': category,
      'imageUrl': imageUrl,
    };
  }

  Medication copyWith({
    int? id,
    String? name,
    String? description,
    double? price,
    int? stock,
    String? type,
    String? category,
    String? imageUrl,
  }) {
    return Medication(
      id: id ?? this.id,
      name: name ?? this.name,
      description: description ?? this.description,
      price: price ?? this.price,
      stock: stock ?? this.stock,
      type: type ?? this.type,
      category: category ?? this.category,
      imageUrl: imageUrl ?? this.imageUrl,
    );
  }
}

File: backend/app/models/medication.py
from datetime import datetime
from app import db

class Category(db.Model):
    """Model for medication categories"""
    __tablename__ = 'categories'
    
    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(100), nullable=False)
    description = db.Column(db.Text)
    medication_type = db.Column(db.String(20), nullable=False)  # 'human' or 'animal'
    
    # Relationships
    medications = db.relationship('Medication', backref='category', lazy=True)
    
    def __repr__(self):
        return f'<Category {self.name}>'

class Medication(db.Model):
    """Model for medications (both human and animal)"""
    __tablename__ = 'medications'
    
    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(100), nullable=False)
    description = db.Column(db.Text)
    full_details = db.Column(db.Text)
    price = db.Column(db.Float, nullable=False)
    stock_quantity = db.Column(db.Integer, default=0)
    medication_type = db.Column(db.String(20), nullable=False)  # 'human' or 'animal'
    category_id = db.Column(db.Integer, db.ForeignKey('categories.id'))
    requires_prescription = db.Column(db.Boolean, default=False)
    dosage_instructions = db.Column(db.Text)
    contraindications = db.Column(db.Text)
    side_effects = db.Column(db.Text)
    storage_instructions = db.Column(db.Text)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    updated_at = db.Column(db.DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    
    # Relationships
    images = db.relationship('MedicationImage', backref='medication', lazy=True, cascade='all, delete-orphan')
    
    def get_thumbnail_url(self):
        """Get the primary image URL or the first available image"""
        primary_image = MedicationImage.query.filter_by(medication_id=self.id, is_primary=True).first()
        if primary_image:
            return primary_image.image_url
            
        # If no primary image, get the first image
        first_image = MedicationImage.query.filter_by(medication_id=self.id).first()
        if first_image:
            return first_image.image_url
            
        # Default image if none available
        return '/static/images/default_medication.jpg'
    
    def __repr__(self):
        return f'<Medication {self.name}>'

class MedicationImage(db.Model):
    """Model for medication images"""
    __tablename__ = 'medication_images'
    
    id = db.Column(db.Integer, primary_key=True)
    medication_id = db.Column(db.Integer, db.ForeignKey('medications.id'), nullable=False)
    image_url = db.Column(db.String(255), nullable=False)
    is_primary = db.Column(db.Boolean, default=False)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    
    def __repr__(self):
        return f'<MedicationImage {self.id} for Medication {self.medication_id}>'